lib/hyperion_test/fake_server/config.rb

```diff
@@ -0,0 +1,36 @@
+class Hyperion
+  class FakeServer
+    class Config
+      # this is passed to the block to allow the caller to configure the fake server
+
+      include Hyperion::Headers
+      include Hyperion::Logger
+
+      def rules
+        @rules ||= []
+      end
+
+      # allow(route)
+      # allow(method, path, headers={})
+      def allow(*args, &handler)
+        rule = allowed_rule(args, handler)
+        rules << rule
+        log_stub(rule)
+      end
+
+      private
+
+      def allowed_rule(args, handler)
+        if args.size == 1 && args.first.is_a?(RestRoute)
+          route = args.first
+          Rule.new(MimicRoute.new(route.method, route.uri.path), route_headers(route), handler, route)
+        else
+          # TODO: deprecate this
+          method, path, headers = args
+          headers ||= {}
+          Rule.new(MimicRoute.new(method, path), headers, handler, nil)
+        end
+      end
+    end
+  end
+end
```

Comment:
Code meant for testing purposes lives in the gem's require list. That means that when another project (ascent-web) uses this gem, ALL the files are loaded whether ascent-web is running in production or test mode. I do not know what the accepted practice is for mocks within a gem. Ideally, it would be loaded in ascent-web only when in development or test mode. The only way that I know how to enforce that would be to put the fake server stuff in another gem that would be included as a 'development' dependency of ascent-web.

Again, I do not know the accepted practice. If we keep fake server here, then we need to be especially aware that all these test files will be loaded when ascent-web is running in production. This also means that we need to be careful about what is automatically 'mocked out' or methods / classes that exist right off the default namespace.